elementary/clients/slack/client.py

```diff
@@ -130,7 +131,11 @@ def _get_channels(self) -> List[dict]:
 
     def _handle_send_err(self, err: SlackApiError, channel_name: str) -> bool:
         err_type = err.response.data['error']
-        if err_type == 'not_in_channel':
+        if err_type == 'ratelimited':
+            rate_limit_waiting_time = int(err.response.headers['Retry-After'])
```

Comment:
add try-except here so we won't fail on trying to parse their retry-after? Also add log that we got rate limit from Slack and therefore waiting X sec (X should be what you get from Slack)

Reply:
Also - can we get rate limit also if a user uses webhook? how should we handle it?

Reply:
You are right about the webhook. Added handler for it as well.

Reply:
I tested it and it seems like there is no rate limit on webhooks.
The error format is different from token and if I can't raise a limit error for webhooks, I can't create the right way to handle it..

Reply:
It seems from the docs that you can get a rate limit also to webhooks - 
<img width="1226" alt="image" src="https://user-images.githubusercontent.com/4517234/188871183-9d87b5c9-8292-4450-b2cf-83aac6f669e1.png">

https://api.slack.com/docs/rate-limits

Seems like we should check the http error code and the headers